llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1039,14 +1073,14 @@ InstructionCost VPWidenIntrinsicRecipe::computeCost(ElementCount VF,
 
   Type *RetTy = ToVectorTy(Ctx.Types.inferScalarType(this), VF);
   SmallVector<Type *> ParamTys;
-  for (unsigned I = 0; I != getNumOperands(); ++I)
+  for (unsigned I = 0; I != NumOperands; ++I)
     ParamTys.push_back(
         ToVectorTy(Ctx.Types.inferScalarType(getOperand(I)), VF));
 
   // TODO: Rework TTI interface to avoid reliance on underlying IntrinsicInst.
   FastMathFlags FMF = hasFastMathFlags() ? getFastMathFlags() : FastMathFlags();
   IntrinsicCostAttributes CostAttrs(
-      VectorIntrinsicID, RetTy, Arguments, ParamTys, FMF,
+      FID, RetTy, Arguments, ParamTys, FMF,
       dyn_cast_or_null<IntrinsicInst>(getUnderlyingValue()));
   return Ctx.TTI.getIntrinsicInstrCost(CostAttrs, CostKind);
```

Comment:
I'm concerned that we still have some VP intrinsics without an implemented cost. 
Could we try to get the cost from `llvm.<intrinsic name>` if `vp.<intrinsic name>` returns Invalid?

Reply:
Cost from llvm.<intrinsic> not <vp.intrinsic>

Reply:
I don't understand. The issue with the cost model should only be that getIntrinsicInstrCost may not have implemented the cost for all vp intrinsics. Alternatively, you could choose one of the vp intrinsics for which getIntrinsicInstrCost has already been implemented as a demo. 
Then we can discuss what to do about invalid costs.

Reply:
Sorry，fixed again~

Reply:
So, why not use vp intrinsic ID directly?

Reply:
some intrinsic's cost will get diffrent result in leagcy model and recipe cost model,eg: llvm.vp.cttz.

Reply:
Is it because the cost of llvm.vp.cttz not implemented yet so you get Invalid, or something else?

Reply:
The cost of vscalble of llvm.cttz is invalid in legacy model
![image](https://github.com/user-attachments/assets/203fc799-fb2b-4d6a-802b-ae8015d9640d)

Reply:
After debugging, I found that ISD::cttz is an extended type and ISD::ctlz is a custom type.I'm trying to solve it
![image](https://github.com/user-attachments/assets/7e6990ba-8b37-4910-9e8a-bbdb1b958ada)